tests/e2e/test/pod-pvc-linode-encryption/chainsaw-test.yaml

```diff
@@ -0,0 +1,125 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/kyverno/chainsaw/main/.schemas/json/test-chainsaw-v1alpha1.json
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  creationTimestamp: null
+  name: pod-pvc-linode-encrypted
+  labels:
+    all:
+    encryptedlinode:
+spec:
+  bindings:
+    - name: nodes
+      # number of nodes in cluster
+      value: ((env('WORKER_NODES') | to_number(@)) + (env('CONTROLPLANE_NODES') | to_number(@)))
+  steps:
+    - name: Check if CSI Driver is deployed
+      try:
+        - assert:
+            file: assert-csi-driver-resources.yaml
+    - name: Create PVC and Pod
+      try:
+        - apply:
+            file: create-pvc-pod.yaml
+      catch:
+        - describe:
+            apiVersion: v1
+            kind: Pod
+        - describe:
+            apiVersion: v1
+            kind: PersistentVolumeClaim
+    - name: Check if Pod is ready and Volume is mounted
+      try:
+        - assert:
+            file: assert-pvc-pod.yaml
+      catch:
+        - describe:
+            apiVersion: v1
+            kind: PersistentVolumeClaim
+        - describe:
+            apiVersion: v1
+            kind: Pod
+    - name: Check if encrypted volume is created and encrypted
+      try:
+        - script:
+            env:
+              - name: TARGET_API
+                value: api.linode.com
+              - name: TARGET_API_VERSION
+                value: v4beta
+              - name: URI
+                value: volumes
+              - name: FILTER
+                value: (to_string({"tags":($namespace)}))
+            content: |
+              set -e
+              RESPONSE=$(curl -s \
+                -H "Authorization: Bearer $LINODE_TOKEN" \
+                -H "X-Filter: $FILTER" \
+                -H "Content-Type: application/json" \
+                "https://api.linode.com/v4beta/volumes")
+              
+              # Check if the results field is 1
+              RESULTS_COUNT=$(echo "$RESPONSE" | jq -r '.results')
+              if [ "$RESULTS_COUNT" -ne 1 ]; then
+              echo "Error: Expected results to be 1, but got $RESULTS_COUNT"
+              exit 1
+              fi
+
+              # Check the encryption field
+              ENCRYPTION_STATUS=$(echo "$RESPONSE" | jq -r '.data[0].encryption')
+              if [ "$ENCRYPTION_STATUS" != "enabled" ]; then
+                echo "Error: Encryption is not enabled (found: $ENCRYPTION_STATUS)"
+                exit 1
+              fi
```

Comment:
For future debugging purpose, can we include the response body when we run into an error?

Reply:
Sure!